### PR TITLE
Serialization transfer.

### PR DESCRIPTION
**Pull request by MonsterFich**
Transferred:

- [x] Serialization base;
- [x] Serialization box;
- [x] Serialization collection;
- [x] Serialization composits;
- [x] Serialization core;
- [x] Serialization integrated types;
- [x] Serialization export file;
- [x] Serialization metadata;
- [x] Serialization optionals;
- [x] Serialization public API;
- [x] Serialization struct;
- [x] Serialization TypeMetaData;
- [x] Text path;
- [x] Text tokenizer;

Also:

- [x] Optimized StructCttiDef;
- [x] Fixed some namespace issues;
- [x] Added missing CTTI file;
- [x] Minor code formatting;

### CODE REVIEW

**Review by Oss1r — APPROVED** [review]

**Comment by nsubaru** [comment]
Please format "\" in Cardinal.CTTI.Minimal.TypeTraits.StructDefintionBuild.h

**Comment by nsubaru** [comment]
Move second argument to next line https://github.com/nsubaru/Cardinal/blob/df223ea93216096d15cc67a05da609239cd2f225/Cardinal/Cardinal.Exceptions.Internals.ixx#L44

File: Cardinal/cpp.hint [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    >(Cardinal::TypeTraits::CreateTypeStr(L"" #Name))
+
+#define DefStructMetaDataObj(Struct, ...)                                          \
+    Cardinal::CTTI::TypeTraits::MakeCustomStruct<Struct>(__VA_ARGS__)
+
+#define DefStructMetaData(Struct, ...)                                             \
+    static constexpr auto FieldMetaData = DefStructMetaDataObj(Struct, __VA_ARGS__) 
+
+
+#define StructCttiDef(StructName, ...)                                     \
```
Comment (nsubaru) [inline Cardinal/cpp.hint]: Format "\" in whole file

**Comment by nsubaru** [comment]
Remove export as it's private part
https://github.com/nsubaru/Cardinal/blob/df223ea93216096d15cc67a05da609239cd2f225/Cardinal/Cardinal.Serialization.TypeMetaRecord.ixx#L514

**Comment by nsubaru** [comment]
Remove anonymous namespace https://github.com/nsubaru/Cardinal/blob/df223ea93216096d15cc67a05da609239cd2f225/Cardinal/Cardinal.Text.Path.ixx#L778

**Comment by nsubaru** [comment]
Remove anonymous namespace
https://github.com/nsubaru/Cardinal/blob/df223ea93216096d15cc67a05da609239cd2f225/Cardinal/Cardinal.Text.Path.ixx#L787

**Review by nsubaru — CHANGES_REQUESTED** [review]
Please fix all comments

**Review by nsubaru — APPROVED** [review]

**Pull request merged**